subworkflows/local/splitncigar/main.nf

```diff
@@ -17,36 +17,51 @@ workflow SPLITNCIGAR {
     main:
     def ch_versions       = Channel.empty()
 
-    def bam_interval = bam.combine(intervals).map{ meta, bam_, bai, intervals_ -> [ meta + [sample:meta.id], bam_, bai, intervals_ ] }
+    def bam_interval = bam
+        .combine(intervals)
+        .map { meta, bam_, bai, intervals_ ->
+            def new_meta = meta + [interval_count:intervals_ instanceof List ? intervals_.size() : 1]
+            [ new_meta, bam_, bai, intervals_ ]
```

Comment:
Why not keep it simple?
```suggestion
            [ meta + [interval_count:intervals_ instanceof List ? intervals_.size() : 1], bam_, bai, intervals_ ]
```

Reply:
I mainly did this for sake of readability on smaller screens. I don't have a strong opinion on this tho so feel free to apply your suggestion if you prefer this :)